wallet/concurrent_test.go

```diff
@@ -0,0 +1,40 @@
+package wallet
+
+import (
+	"sync"
+	"testing"
+)
+
+func TestConcurrentSafe(t *testing.T) {
+
+	w := Wallet{}
+
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for j := 0; j < 100; j++ {
+				err := w.Deposit(Bitcoin(1))
+				if err != nil {
+					t.Errorf("Wrong error: %v", err)
+				}
+			}
+
+			for j := 0; j < 50; j++ {
+				err := w.Withdraw(Bitcoin(1))
+				if err != nil {
+					t.Errorf("Wrong error: %v", err)
+				}
+			}
```

Comment:
let's split `Deposit` and `Withdraw` calls into separate goroutines, that way we'll potentially get more overlap during execution.